src/js/sequences/components/Genome/EditGenomeSequence.tsx

```diff
@@ -0,0 +1,182 @@
+import { DialogPortal } from "@radix-ui/react-dialog";
+import { Field, Form, Formik, FormikErrors, FormikTouched } from "formik";
+import { find } from "lodash-es";
+import React from "react";
+import { connect } from "react-redux";
+import { pushState } from "../../../app/actions";
+import { Dialog, DialogContent, DialogOverlay, DialogTitle, SaveButton } from "../../../base";
+import { getError } from "../../../errors/selectors";
+import PersistForm from "../../../forms/components/PersistForm";
+import { editSequence } from "../../../otus/actions";
+import { getActiveIsolateId, getOTUDetailId } from "../../../otus/selectors";
+import { OTUSegment } from "../../../otus/types";
+import { routerLocationHasState } from "../../../utils/utils";
+import { getActiveSequence, getUnreferencedSegments } from "../../selectors";
+import { SequenceForm, validationSchema } from "../SequenceForm";
+import SegmentField from "./SegmentField";
+
+function getInitialValues({ initialSegment, initialAccession, initialDefinition, initialHost, initialSequence }) {
+    return {
+        segment: initialSegment || "",
+        accession: initialAccession || "",
+        definition: initialDefinition || "",
+        host: initialHost || "",
+        sequence: initialSequence || "",
+    };
+}
+
+export function castValues(segments: OTUSegment[]) {
+    return function (values: formValues) {
+        const segment = find(segments, { name: values.segment }) ? values.segment : "";
+        return { ...values, segment };
+    };
+}
+
+type formValues = {
+    segment: string;
+    accession: string;
+    definition: string;
+    host: string;
+    sequence: string;
+};
+
+type EditGenomeSequenceProps = {
+    initialAccession: string;
+    initialDefinition: string;
+    initialHost: string;
+    initialSegment: string;
+    initialSequence: string;
+    /** A list of unreferenced segments */
+    segments: OTUSegment[];
+    id: string;
+    isolateId: string;
+    otuId: string;
+    /** Indicates whether the dialog for editing a sequence is visible */
+    show: boolean;
+    /** A callback function to hide the dialog */
+    onHide: () => void;
+    /** A callback function to update the sequence */
+    onSave: (
+        otuId: string,
+        isolateId: string,
+        sequenceId: string,
+        accession: string,
+        definition: string,
+        host: string,
+        segment: string,
+        sequence: string,
+    ) => void;
+};
+
+/**
+ * Displays dialog to edit a genome sequence
+ */
+export function EditGenomeSequence({
+    initialAccession,
+    initialDefinition,
+    initialHost,
+    initialSegment,
```

Comment:
```suggestion
    initialSegmentName,
```
I think there is some annoying ambiguity here where segment can mean either the name of the segment or the complete segment objects like in `segments`. Open to other solns